File: Day96/maxPathSum.js
class Solution 
{   constructor()
    {
        this.res = -999999999;
    }
    //Function to return maximum path sum from any node in a tree.
    
    fill(root)
    {
        if(!root)return 0;
        
        let l = Math.max(0,this.fill(root.left));
        let r = Math.max(0,this.fill(root.right));
        
        this.res = Math.max(this.res, l + r +root.data);
        
        return root.data + Math.max(l,r);
        
        
    }
    
    findMaxSum(root)
    {
        this.fill(root);
        return this.res;
    }
}

File: Day142/maxPathSum.cpp
class Solution {
  public:
    //Function to return maximum path sum from any node in a tree.
    int findMaxSum(Node* root)
    {
        if(!root)return 0;
        int res = INT_MIN;
        fill(root,res);
        return res;
    }
    
    int fill(Node* root,int& res)
    {
        if(!root)return 0;
        
        int l = max(0,fill(root->left,res));
        int r = max(0,fill(root->right,res));
        
        res = max(res,l+r+root->data);
        
        return max(l,r)+root->data;
    }
};